### PR TITLE
docs: Authentication for Git repositories

### PR DESCRIPTION
**Pull request by max-cx**
## What issues does this pull request fix or reference
RHDEVDOCS-3791
https://github.com/eclipse/che/issues/21298

## Pull request checklist

The author and the reviewers validate the content of this pull request with the following checklist, in addition to the [automated tests](code_review_checklist.adoc).

- Any procedure:
  - [ ] Successfully tested.
- Any page or link rename:
  - [ ] The page contains a redirection for the previous URL.
  - Propagate the URL change in:
    - [ ] Dashboard [default branding data](https://github.com/eclipse-che/che-dashboard/blob/main/packages/dashboard-frontend/src/services/bootstrap/branding.constant.ts)
    - [ ] Chectl [constants.ts](https://github.com/che-incubator/chectl/blob/master/src/constants.ts)
- [ ] Builds on [Eclipse Che hosted by Red Hat](https://workspaces.openshift.com).
- [ ] the *`Validate language on files added or modified`* step reports no vale warnings.

### CODE REVIEW

File: modules/end-user-guide/pages/authentication-for-git-repositories.adoc [hunk 1]
```diff
@@ -0,0 +1,18 @@
+:_content-type: CONCEPT
+:description: Authentication for Git repositories
+:keywords: authentication, authenticate, github, gitlab, bitbucket
+:navtitle: Authentication for Git repositories
+// :page-aliases:
+
+[id="authentication-for-git-repositories_{context}"]
+= Authentication for Git repositories
+
+To be written.
```
Comment (max-cx) [inline modules/end-user-guide/pages/authentication-for-git-repositories.adoc:10]:
```suggestion
To be written.
```

Comment (dkwon17) [inline modules/end-user-guide/pages/authentication-for-git-repositories.adoc]:
```suggestion
To be written later.
```

File: modules/end-user-guide/pages/user-onboarding.adoc [hunk 2]
```diff
@@ -0,0 +1,15 @@
+:_content-type: CONCEPT
+:description: User onboarding
+:keywords: getting-started, user-onboarding, new-user, new-users
+:navtitle: User onboarding
+// :page-aliases:
+
+[id="user-onboarding_{context}"]
+= User onboarding
+
+If your organization is already running a {prod-short} instance, you can get started as a new user of {prod-short} by first learning how to start a new workspace and how to manage your workspaces:
```
Comment (max-cx) [inline modules/end-user-guide/pages/user-onboarding.adoc]: This is content from https://github.com/eclipse-che/che-docs/pull/2252

**Comment by dkwon17** [comment]
I have added to the draft: 
image

File: modules/end-user-guide/pages/authentication-for-git-repositories.adoc [hunk 3]
```diff
@@ -0,0 +1,16 @@
+:_content-type: CONCEPT
+:description: Authentication for Git repositories
+:keywords: authentication, authenticate, github, gitlab, bitbucket
+:navtitle: Authentication for Git repositories
+// :page-aliases:
+
+[id="authentication-for-git-repositories_{context}"]
+= Authentication for Git repositories
+
+Setting up authentication to a Git server with your GitHub, GitLab or BitBucket account removes the need to provide your Git credentials to `git` when performing operations such as cloning private repositories and pushing to remote repositories.
```
Comment (max-cx) [inline modules/end-user-guide/pages/authentication-for-git-repositories.adoc:10]:
```suggestion
Setting up authentication to a Git server with your GitHub, GitLab or BitBucket account removes the need to provide your Git credentials to Git when performing operations such as cloning private repositories and pushing to remote repositories.
```

**Comment by max-cx** [comment]
Preview:

image

image

File: modules/end-user-guide/pages/user-onboarding.adoc [hunk 4]
```diff
@@ -7,8 +7,9 @@
 [id="user-onboarding_{context}"]
 = User onboarding
 
-If your organization is already running a {prod-short} instance, you can get started as a new user of {prod-short} by first learning how to start a new workspace and how to manage your workspaces:
+If your organization is already running a {prod-short} instance, you can get started as a new user by learning how to start a new workspace, manage your workspaces, and authenticate yourself to remote Git repositories from workspaces:
 
 . xref:starting-a-new-workspace-with-a-clone-of-a-git-repository.adoc[]
 . xref:optional-parameters-for-the-urls-for-starting-a-new-workspace.adoc[]
 . xref:basic-actions-you-can-perform-on-a-workspace.adoc[]
+. xref:authenticating-yourself-to-a-git-server-from-a-workspace.adoc[]
```
Comment (Srivaralakshmi) [inline modules/end-user-guide/pages/user-onboarding.adoc:15]: Suggestion: The xref says authenticate yourself to a git server and the lead sentence says remote Git repositories. Stick to one type of usage for consistency.

**Review by Srivaralakshmi — COMMENTED** [review]
@max-cx Left a comment. Nitpick. Otherwise, LGTM. Thanks!

File: modules/end-user-guide/pages/user-onboarding.adoc (hunk 4 above)
Comment (max-cx) [reply modules/end-user-guide/pages/user-onboarding.adoc:15]: Thank you so much for catching this, Sri!

**Review by Srivaralakshmi — APPROVED** [review]
@max-cx Thanks for the changes. Approved!

**Review by dkwon17 — APPROVED** [review]

**Review by dkwon17 — APPROVED** [review]
Other than failing build, LGTM

File: modules/end-user-guide/pages/authenticating-yourself-to-a-git-server-from-a-workspace.adoc [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+To configure user authentication to a Git server from a workspace, {prod-short} provides two options:
+
+* Your administrator sets up an xref:administration-guide:oauth-for-github-gitlab-or-bitbucket.adoc[OAuth application on GitHub, GitLab, or Bitbucket] for your organization's {prod} instance.
+
+* You create your own, user xref:creating-a-kubernetes-secret-for-a-git-credentials-store.adoc[{kubernetes} Secret for a Git credentials store].
+
+.Additional resources
+* xref:administration-guide:oauth-for-github-gitlab-or-bitbucket.adoc[Administration Guide: OAuth for GitHub, GitLab, or Bitbucket]
+* xref:creating-a-kubernetes-secret-for-a-git-credentials-store.adoc[User Guide: Creating a Kubernetes Secret for a Git credentials store]
```
Comment (themr0c) [inline modules/end-user-guide/pages/authenticating-yourself-to-a-git-server-from-a-workspace.adoc]: ERROR (asciidoctor): target of xref not found: creating-a-kubernetes-secret-for-a-git-credentials-store.adoc

File: modules/end-user-guide/pages/authenticating-yourself-to-a-git-server-from-a-workspace.adoc [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+[id="authenticating-yourself-to-a-git-server-from-a-workspace_{context}"]
+= Authenticating yourself to a Git server from a workspace
+
+In a workspace, you can run Git commands that require user authentication like cloning a remote private Git repository or pushing to a remote public or private Git repository.
+
+To configure user authentication to a Git server from a workspace, {prod-short} provides two options:
+
+* Your administrator sets up an xref:administration-guide:oauth-for-github-gitlab-or-bitbucket.adoc[OAuth application on GitHub, GitLab, or Bitbucket] for your organization's {prod} instance.
+
+* You create your own, user xref:creating-a-kubernetes-secret-for-a-git-credentials-store.adoc[{kubernetes} Secret for a Git credentials store].
```
Comment (themr0c) [inline modules/end-user-guide/pages/authenticating-yourself-to-a-git-server-from-a-workspace.adoc]: ERROR (asciidoctor): target of xref not found: creating-a-kubernetes-secret-for-a-git-credentials-store.adoc

**Pull request merged**